### PR TITLE
Fix less than sign

### LINKED ISSUE #11: Syntax highlighting issue when there is no space preceding <
Ported from https://github.com/Microsoft/TypeScript-Sublime-Plugin/issues/285

Related to https://github.com/Microsoft/TypeScript-Sublime-Plugin/issues/265.

Issue:
image

Correct:
image

### PR DESCRIPTION
**Pull request by zhengbli**
Fixes #11 

In places like interface / class declaration, type annotation, type alias declaration, we are sure that the `<` sign can only be interpreted as a generic parameter start, therefore I added a new classification called `known-type-parameter` in these places to avoid the ambiguity. 

In other places, the `<` relies on the existence of `>\s*(` later in the line to tell if it is a less-than sign or a generic start.

### CODE REVIEW

File: TypeScriptReact.YAML-tmLanguage [hunk 1]
```diff
@@ -114,7 +114,7 @@ repository:
     - include: '#comment'
     - include: '#object-heritage' # Must be before object-name to consume the "extends" keyword.
     - include: '#object-name'
-    - include: '#type-parameters' # Must be applied after object-name e.g. interface KeyValuePair<K, V>
+    - include: '#know-type-parameters' # Must be applied after object-name e.g. interface KeyValuePair<K, V>
```
Comment (anubmat) [inline TypeScriptReact.YAML-tmLanguage]: I think you misspelt it here. '#know-type-parameters'

**Comment by aeschli** [comment]
We've been using the proposed changes for several weeks now in VSCode. All looks good for js (using TypeScriptReact grammar) and ts.
Tsx has issues, see https://github.com/Microsoft/TypeScript-TmLanguage/issues/202
I reverted tsx in VSCode to the version we had in 1.2 (current master, https://github.com/Microsoft/TypeScript-TmLanguage/commit/a78f05a1af40a3dbc1a87992b37e34e69631c94d).

**Comment by anubmat** [comment]
@zhengbli.. Do you want to check this in? 👍

**Comment by zhengbli** [comment]
@anubmat I'll try to see if #202 can be fixed with an update, otherwise I will merge soon. Thanks.

**Comment by zhengbli** [comment]
@aeschli @anubmat This new update should address #202. Please let me know if you found anything else. Thanks!

**Comment by zhengbli** [comment]
Gently pinging @aeschli

**Comment by sheetalkamat** [comment]
@zhengbli  you shouldn't merge this as I have already worked this out little differently in the updated grammar I have been working.

**Comment by zhengbli** [comment]
Closing.

**Pull request closed without merging**